packages/@aws-cdk/aws-apigatewayv2-integrations/README.md

```diff
@@ -149,6 +149,57 @@ const httpEndpoint = new HttpApi(stack, 'HttpProxyPrivateApi', {
 });
 ```
 
+### AWS Service Integrations
+
+AWS Service integrations allow for API Gateway to integrate directly with the following AWS Services:
+
+- EventBridge
+  - PutEvents
+- SQS
+  - SendMessage
+  - ReceiveMessage
+  - DeleteMessage
+  - PurgeQueue
+- Kinesis
+  - PutRecord
+- Step Functions
+  - StartExecution
+  - StartSyncExecution
+  - StopExecution
+
+The following code configures a `message` route, which creates an SQS message containing the request body:
+
+```ts
+const queue = new Queue(stack, 'Queue');
+const httpApi = new HttpApi(stack, 'IntegrationApi');
+
+const role = new Role(stack, 'SQSRole', {
+  assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
+});
+role.addToPrincipalPolicy(new PolicyStatement({
+  actions: ['sqs:*'],
+  resources: [queue.queueArn],
+}));
+
+httpApi.addRoutes({
+  path: '/message',
+  methods: [HttpMethod.POST],
+  integration: new SqsSendMessageIntegration({
+    role,
+    body: StringMappingExpression.fromMapping(Mapping.fromRequestBody()),
+    queue: QueueMappingExpression.fromQueue(queue),
+  }),
+});
+```
+
+Integrations should always specify a role, with appropriate permissions to allow the actions.
```

Comment:
Why can't the integration automatically add the correct permissions to the role?

Reply:
Hi, thanks for the review. I thought about that, but since the queue name, for example, could come from the payload, it wouldn't be possible for the cdk to know what permissions may be required.

Reply:
Got it. But we could maybe identify when the queue name is known at synth time and in this case give the right permissions.